annotated_test.go

```diff
@@ -477,6 +480,32 @@ func TestAnnotatedAs(t *testing.T) {
 				assert.Equal(t, s.String(), "another stringer")
 			},
 		},
+		{
+			desc: "value type convertible to target type",
+			provide: fx.Provide(
+				fx.Annotate(func() myProvideFunc {
+					return func() string {
+						return "provide func example"
+					}
+				}, fx.As(new(myInvokeFunc))),
+			),
+			invoke: func(h myInvokeFunc) {
+				assert.Equal(t, "provide func example", h())
+			},
+		},
+		{
+			desc: "anonymous value type convertible to target type",
```

Comment:
should we add a test case for a function vaoue that is not convertible to the target type?